### ISSUE TITLE
DataStore query with predicate with AWSDateTime value returns incorrect result

### ISSUE THREAD

**Issue by HuiSF** [aws-amplify/amplify-flutter#1245]
**Describe the bug**

Take schema
```graphql
type Blog @model {
  name: String!
  awsDateTime: AWSDateTime
}
```

Save records of `Blog`

```dart
var records = [
  Blog(
    name: 'time 1',
    awsDateTime: TemporalDateTime(DateTime(2020, 1, 1, 11, 30, 45))
  ),
  Blog(
    name: 'time 1',
    awsDateTime: TemporalDateTime(DateTime(2020, 1, 1, 11, 30, 45, 100))
  ),
  Blog(
    name: 'time 1',
    awsDateTime: TemporalDateTime(DateTime(2020, 1, 1, 11, 30, 45, 100, 250)),
  )
];

for (var record in records) {
  await Amplify.DataStore.save(record);
}
```

The `awsDateTime` values are stored in local SQLite DB as the following screenshot

<img width="186" alt="Screen Shot 2022-01-04 at 12 25 39 PM" src="https://user-images.githubusercontent.com/10602282/148119500-a1b150e5-85aa-433f-8f85-c77d0a2dc396.png">
OS and Android.  This issue is reproducible in (check all that apply):

Note that all values are in different format.

Run below query:

```dart
var blogs = await Amplify.DataStore.query(Blog.classType,
    where: Blog.AWSDATETIME.le(
        TemporalDateTime.fromString('2020-01-01T19:30:45.100000000Z')));
print(blogs);
```
Which returns
```
flutter: Blog {id=61ccf677-da48-4246-abcc-731c57c7fab8, name=blog 2, awsDateTime=2020-01-01T19:30:45.100000000Z}
flutter: Blog {id=25c3bf0a-8ccc-44c4-895e-f1b2f414f20b, name=blog 4, awsDateTime=2020-01-01T19:30:45+01:00}
```

Where the expected results should contains only `blog 1` and `blog 2`.

**To Reproduce**

**Expected behavior**
Query with predicate with AWSDateTime value should return correct results.

**Screenshots**

**Platform**
Amplify Flutter current supports
[x] Android
[x] iOS

<details>

  <summary>Output of <code>flutter doctor -v</code></summary>
  
  ```Paste your output of "flutter doctor -v" here```

</details>

<details>

  <summary>Dependencies (<code>pubspec.lock</code>)</summary>
  
  ```
  Paste the contents of your "pubspec.lock" file here
  ```

</details>

**Smartphone (please complete the following information):**

**Additional context**

**Comment by BramDriesen** [comment]
@HuiSF Could you perhaps try an order descending on your example and see what happens?

**Comment by HuiSF** [comment]
Hi @BramDriesen thanks for following up, I will take a look. 
Could you send me some samples of the DataTime values that you may use?

**Comment by HuiSF** [comment]
I did a quick run, sort seems working as expected on `AWSDateTime` fields.
<img width="891" alt="image" src="https://user-images.githubusercontent.com/10602282/154747607-6a202b3a-ca41-4129-9d08-54c2e33019e2.png">

Notices that screenshot shown datetime values are padded right with 0s.

Can you check @BramDriesen , when you run query, how does the value of the AWSDateTime field look like?

**Comment by BramDriesen** [comment]
Hi @HuiSF

This is strange, it indeed now shows the dates in the correct order:
<img width="337" alt="image" src="https://user-images.githubusercontent.com/12573202/154748571-f8d6f37f-f2fa-4f16-ab66-7bb2e14b95ce.png">

(it didn't do that before).

So I guess that now my issue lays with the `GroupedListView` widget rather as with the data. 😅 

Sorry for wasting your time a bit here, but thanks so much for the effort!!

**Comment by HuiSF** [comment]
Fix has been release with amplify-flutter 0.4.0.
